runtime/shared_common/CacheMap.cpp

```diff
@@ -6421,6 +6603,51 @@ SH_CacheMap::appendCacheDescriptorList(J9VMThread* currentThread, J9SharedClassC
 
 #endif /* J9SHR_CACHELET_SUPPORT */
 
+/**
+ * Allocate, initialise and append a cache descriptor to the end of descriptor list.
+ * @param[in] currentThread The current VMThread
+ * @param[in] sharedClassConfig The J9SharedClassConfig struct
+ * @param[in] ccToUse The SH_CompositeCacheImpl that the descriptor represents
+ *
+ * @@return Address of new J9SharedClassCacheDescriptor or NULL on failure.
+ */
+J9SharedClassCacheDescriptor*
+SH_CacheMap::appendCacheDescriptorList(J9VMThread* currentThread, J9SharedClassConfig* sharedClassConfig, SH_CompositeCacheImpl* ccToUse)
+{
+	PORT_ACCESS_FROM_VMC(currentThread);
+	J9SharedClassCacheDescriptor *cacheDesc;
+
+	cacheDesc = (J9SharedClassCacheDescriptor *)j9mem_allocate_memory(sizeof(J9SharedClassCacheDescriptor), J9MEM_CATEGORY_CLASSES);
+	if (!cacheDesc) {
+		return NULL;
+	}
+	memset(cacheDesc, 0, sizeof(J9SharedClassCacheDescriptor));
+
+	if (sharedClassConfig->configMonitor) {
+		enterLocalMutex(currentThread, sharedClassConfig->configMonitor, "config monitor", "initializeROMSegmentList");
```

Comment:
Please pass in the correct function name, here and calling exitLocalMutex() below,

Reply:
Fixed